### PR TITLE
Allow tornado pieces to be injected into HTTP inbound

### PR DESCRIPTION
**Pull request by breerly**
This will ensure users don't get "locked out" when it comes to extending/composing.

Basically, I was unable to add a `/health` endpoint for Larr' :P

@yarpc/python

### CODE REVIEW

**Comment by junchaowu** [comment]
what's the difference between tornado http and web application?

**Comment by breerly** [comment]
@junchaowu httpserver gives us access to the low level socket options

File: yarpc/transport/http/inbound.py [hunk 1]
```diff
@@ -32,19 +33,32 @@
...
 class HTTPInbound(object):
 
-    def __init__(self, port, app=None):
+    TORNADO_APP_CLASS = tornado.web.Application
+    HTTP_SERVER_CLASS = tornado.httpserver.HTTPServer
+
+    def __init__(self, port, app=None, http_server_cls=None):
         self._port = port
         self._app = app
+        self._http_server_cls = http_server_cls
```
Comment (abhinav) [inline yarpc/transport/http/inbound.py]: Why not apply the default here instead of in `start`

Comment (breerly) [inline yarpc/transport/http/inbound.py]: yeah sure

**Comment by abhinav** [comment]
LGTM

**Pull request merged**